ls1intum/Artemis
# `Adaptive learning`: Allow instructors to participate in learning paths

### Checklist
#### General


#### Server


### Motivation and Context


When navigating to the learning path participation view as an instructor, the server throws a forbidden access error.

### Description

Allow all users that are at least a student in the course to access the participation view.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with at least one Competency linked at least one lecture (e.g. https://artemis-test1.artemis.cit.tum.de/courses/24/exercises)

1. Log in to Artemis
2. Navigate to Course (e.g. https://artemis-test1.artemis.cit.tum.de/courses/24/exercises)
3. Click on the Learning Path tab
4. You should see no alerts (Access Forbidden)


### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [LearningPathResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5752/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/LearningPathResource.html) | 95% | ✅ |

 
